src/frontend/tests/e2e-playwright/dashboard/recoveryPhrase.spec.ts

```diff
@@ -0,0 +1,257 @@
+import { expect, Locator, Page, test } from "@playwright/test";
+import {
+  createNewIdentityInII,
+  dummyAuth,
+  II_URL,
+  readClipboard,
+} from "../utils";
+
+class RecoveryPhraseWizard {
+  #dialog: Locator;
+
+  constructor(dialog: Locator) {
+    this.#dialog = dialog;
+  }
+
+  async acknowledge(): Promise<void> {
+    await expect(
+      this.#dialog.getByRole("heading", { name: "Before you continue" }),
+    ).toBeVisible();
+    await this.#dialog.getByRole("checkbox", { name: "I acknowledge" }).check();
+    await this.#dialog.getByRole("button", { name: "Continue" }).click();
+  }
+
+  async writeDown(): Promise<string[]> {
+    await expect(
+      this.#dialog.getByRole("heading", { name: "Save your recovery phrase" }),
+    ).toBeVisible();
+    await this.#dialog.getByRole("button", { name: "Click to reveal" }).click();
+    // We use selection and copying to the clipboard here to make sure that it
+    // works correctly for power-users that don't physically write things down.
+    await this.#dialog.getByRole("list").selectText();
+    await this.#dialog.press("Meta+c");
+    const clipboard = await readClipboard(this.#dialog.page());
+    await this.#dialog
+      .getByRole("button", { name: "I have written it down" })
+      .click();
+    return clipboard.trim().split("\n");
+  }
+
+  async verify(words: string[]): Promise<void> {
+    await expect(
+      this.#dialog.getByRole("heading", {
+        name: "Verify your recovery phrase",
+      }),
+    ).toBeVisible();
+    for (const word of words) {
+      // Recovery phrase could have duplicate words, so always
+      // select the first enabled button that isn't pressed.
+      await this.#dialog
+        .getByRole("list")
+        .getByRole("button", { name: word, disabled: false, pressed: false })
+        .first()
+        .click();
+    }
+  }
+
+  async reset(): Promise<void> {
+    await expect(
+      this.#dialog.getByRole("heading", {
+        name: "Reset your recovery phrase?",
+      }),
+    ).toBeVisible();
+    await this.#dialog.getByRole("button", { name: "Reset" }).click();
+  }
+}
+
+class RecoveryPhrasePage {
+  #page: Page;
+
+  constructor(page: Page) {
+    this.#page = page;
+  }
+
+  async activate(options?: { skipVerification?: boolean }): Promise<string[]> {
```

Comment:
Nit - this function is called "activate," but shouldn't it be called, e.g., `check_activation_succeeds`? It seems to be starting with an expectation that the page in question has already been activated.